src/dune_rules/pkg_rules.ml

```diff
@@ -1739,8 +1848,11 @@ let lock_dir_active = Lock_dir.lock_dir_active
 let lock_dir_path = Lock_dir.get_path
 
 let exported_env context =
-  let+ all_packages = all_packages context in
-  let env = Pkg.build_env_of_deps all_packages in
+  let* all_packages = all_packages context in
+  let+ env =
+    Compiler.of_context context
+    >>| Compiler.add_path_to_env ~env:(Pkg.build_env_of_deps all_packages)
```

Comment:
Order is wrong here too. The fact that there are two places where we must keep the order in mind is concerning.